### ISSUE TITLE
Since update to 18.0.4, I cannot validate invoices

### ISSUE THREAD

**Issue by xentity** [Dolibarr/dolibarr#27254]
### Bug

After hitting the button "validate" on the page of the draft invoice, the main part of the website remain blank. The left and the top navigation appears, but nothing else.

![image](https://github.com/Dolibarr/dolibarr/assets/6489407/de9f23c6-89cd-44f3-80ed-5f0319584159)
 

### Environment Version

18.0.4

### Environment OS

Ubuntu 22.04.3 LTS

### Environment Web server

Apache 2.4.52

### Environment PHP

8.1.2

### Environment Database

MariaDB 10.6.12

### Environment URL(s)

/dolibarr/compta/facture/card.php?facid=xyz&action=valid

### Expected and actual behavior

Normally, a popup appears, if the invoice should be validated. But there is nothing...

The behavior happens with a normal user as well as doliadmin. I changed several things (payment, template, products, owner), but there was no indicator of a root cause. Even the Firefox DevTools show no errors.

Nothing appears in /var/log/apache2/error.log, no obvious errors in /var/www/html/dolibarr/documents/dolibarr.log

### Steps to reproduce the behavior

Create a new invoice, add some lines and try to validate.

**Comment by erikvanberkum** [comment]
Can you check the user persmissions (refresh them), I had some issues in the past refreshing them solved the problem.

**Comment by xentity** [comment]
> Can you check the user persmissions (refresh them), I had some issues in the past refreshing them solved the problem.

What do you mean? Take the permission and re-assign them? Or is there a general command for that?

**Comment by erikvanberkum** [comment]
goto /user/home.php?leftmenu=users
select your own user name or when that is not shown 
goto /user/perms.php?id=1

![image](https://github.com/Dolibarr/dolibarr/assets/7335299/5050d7b9-1509-4562-856e-c5f813277816)


toggle your user permissions on and off

**Comment by erikvanberkum** [comment]
I just did my upgrade to version 18.0.4 from 18.0.2 and dont have any problems

**Comment by xentity** [comment]
> I just did my upgrade to version 18.0.4 from 18.0.2 and dont have any problems

I did an upgrade from 16.0.2 to 18.0.4, it occurred with two installations.

**Comment by xentity** [comment]
> toggle your user permissions on and off

Below "invoices" I have 5 permissions only. I removed all invoice permissions and re-granted them, but no changes. It is strange that even the admin user is effected.

**Comment by erikvanberkum** [comment]
What happens if you disable the invoice module and reenable it?

**Comment by xentity** [comment]
I guess I found the mistake. We have our own invoice module (mod_facture_venus). I watched the logs on the QS system (lone user) during the logs carefully.

There was an error with mb_subsstr():
`$posnumstart = mb_strrpos($maskwithnocode, $maskcounter, 'UTF-8');` 
and a string instead of int given for the offset. With PHP8.0 the order of the args was changed... 
`$posnumstart = mb_strrpos($maskwithnocode, $maskcounter, 1, 'UTF-8');`
did the trick. 

I excluded this module as root cause since the invoice draft was generated, but these seems to be different files.
